# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/libglvnd/vendor-dlopen-diagnostics.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- libglvnd-v1.7.0/src/EGL/libeglvendor.c.vanilla	2023-09-12 16:02:00.000000000 +0200
+++ libglvnd-v1.7.0/src/EGL/libeglvendor.c	2025-12-14 12:00:17.474090739 +0100
@@ -520,6 +520,7 @@
 
     vendor->dlhandle = dlopen(dlopenName, RTLD_LAZY);
     if (vendor->dlhandle == NULL) {
+        fprintf(stderr, "WARNING: Could not load EGL vendor library %s: %s\n", filename, dlerror());
         goto fail;
     }
 
--- libglvnd-v1.7.0/src/GLX/libglxmapping.c.vanilla	2023-09-12 16:02:00.000000000 +0200
+++ libglvnd-v1.7.0/src/GLX/libglxmapping.c	2025-12-14 12:02:04.753524049 +0100
@@ -427,10 +427,12 @@
             if (filename) {
                 vendor->dlhandle = dlopen(filename, RTLD_LAZY);
             }
-            free(filename);
             if (vendor->dlhandle == NULL) {
+                fprintf(stderr, "WARNING: Could not load GLX vendor library %s: %s\n", filename, dlerror());
+                free(filename);
                 goto fail;
             }
+            free(filename);
 
             glxMainProc = dlsym(vendor->dlhandle, __GLX_MAIN_PROTO_NAME);
             if (!glxMainProc) {
